iot-management-device/src/test/kotlin/com/example/iotmanagementdevice/kafka/KafkaTest.kt

```diff
@@ -0,0 +1,85 @@
+package com.example.iotmanagementdevice.kafka
+
+import DeviceFixture.createDevice
+import DeviceFixture.createDeviceUpdateRequestDto
+import com.example.internal.KafkaTopic.KafkaDeviceUpdateEvents.NOTIFY
+import com.example.internal.commonmodels.DeviceUpdateNotification
+import com.example.iotmanagementdevice.repository.AbstractMongoTest
+import com.example.iotmanagementdevice.service.device.DeviceService
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.common.serialization.ByteArrayDeserializer
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.awaitility.Awaitility.await
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Import
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate
+import reactor.core.scheduler.Schedulers
+import reactor.kafka.receiver.KafkaReceiver
+import reactor.kafka.receiver.ReceiverOptions
+import java.util.concurrent.TimeUnit
+
+@Import(KafkaTest.KafkaTestConfiguration::class)
+class KafkaTest : AbstractMongoTest {
+    @Autowired
+    private lateinit var deviceService: DeviceService
+
+    @Autowired
+    private lateinit var testConsumer: KafkaReceiver<String, ByteArray>
```

Comment:
will int work if you write many tests that use testConsumer?

Reply:
if tests run in parallel it will not work, fix it by creating a new consumer group for every new test